app/src/main/java/org/mozilla/tv/firefox/hint/HintBinder.kt

```diff
@@ -37,12 +38,29 @@ object HintBinder {
                                 .start()
                     }
                     .subscribe { shouldDisplay ->
-                        hintContainer.isVisible = true
                         val translationY = when (shouldDisplay) {
                             true -> 0f
                             false -> hintContainer.height.toFloat()
                         }
+
                         hintContainer.animate()
+                                // If the hint bar is appearing, set the hintContainer.isVisible
+                                // to true before the animation begins. If the hint bar is
+                                // disappearing, set the hintContainer.isiVisible to false after
+                                // the animation ends.
+                                .setListener(object : Animator.AnimatorListener {
```

Comment:
nit: you can use `AnimatorListenerAdapter` so you don't have to override `onAnimationCancel/Repeat` (note: you don't need to call super either in the other methods).